src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/VtableIndexStubGenerator.cs

```diff
@@ -288,48 +351,48 @@ private static IncrementalStubGenerationContext CalculateStubInformation(MethodD
             INamedTypeSymbol? iUnmanagedInterfaceTypeInstantiation = symbol.ContainingType.AllInterfaces.FirstOrDefault(iface => SymbolEqualityComparer.Default.Equals(iface.OriginalDefinition, iUnmanagedInterfaceTypeType));
             if (iUnmanagedInterfaceTypeInstantiation is null)
             {
-                // Report invalid configuration
+                // TODO: Report invalid configuration
             }
             else
             {
-                typeKeyType = ManagedTypeInfo.CreateTypeInfoForTypeSymbol(iUnmanagedInterfaceTypeInstantiation.TypeArguments[0]);
+                // The type key is the second generic type parameter, so we need to get the info for the
+                // second argument.
+                typeKeyType = ManagedTypeInfo.CreateTypeInfoForTypeSymbol(iUnmanagedInterfaceTypeInstantiation.TypeArguments[1]);
             }
 
+            MarshallingInfo exceptionMarshallingInfo = CreateExceptionMarshallingInfo(virtualMethodIndexAttr, environment.Compilation, generatorDiagnostics, virtualMethodIndexData.ExceptionMarshalling, virtualMethodIndexData.ExceptionMarshallingCustomType);
```

Comment:
Should we also have a check/diagnostic for invalid `ExceptionMarshalling` + `ExceptionMarshallingCustomType` combination?